toolchain/check/import_ref.cpp

```diff
@@ -755,10 +764,23 @@ class ImportRefResolver {
                            SemIR::ConstantId class_const_id) -> ResolveResult {
     const auto& import_class = import_ir_.classes().Get(inst.class_id);
 
-    // On the first pass, create a forward declaration of the class for any
-    // recursive references.
+    SemIR::ClassId class_id = SemIR::ClassId::Invalid;
     if (!class_const_id.is_valid()) {
-      class_const_id = MakeIncompleteClass(import_class);
+      // On the first pass, create a forward declaration of the class for any
+      // recursive references.
+      std::tie(class_id, class_const_id) = MakeIncompleteClass(import_class);
+    } else {
+      // On the second pass, compute the class ID from the constant value of the
+      // declaration.
+      if (auto class_type = context_.insts().TryGetAs<SemIR::ClassType>(
+              class_const_id.inst_id())) {
+        class_id = class_type->class_id;
+      } else {
+        auto generic_class_type =
+            context_.types().GetAs<SemIR::GenericClassType>(
+                context_.insts().Get(class_const_id.inst_id()).type_id());
+        class_id = generic_class_type.class_id;
+      }
```

Comment:
It's probably better to only read the instruction once.

```suggestion
      auto class_const_inst = context_.insts().Get(class_const_id.inst_id());
      if (auto class_type = class_const_inst.TryAs<SemIR::ClassType>()) {
        class_id = class_type->class_id;
      } else {
        auto generic_class_type =
            context_.types().GetAs<SemIR::GenericClassType>(class_const_inst.type_id());
        class_id = generic_class_type.class_id;
      }
```

Reply:
I think / hope that it shouldn't matter -- the compiler should be smart enough to see the array can't have changed between the two accesses -- but it definitely can't hurt to help it out :) Done.